kratos/python_scripts/gid_output_process.py

```diff
@@ -275,6 +275,8 @@ def IsOutputStep(self):
     def PrintOutput(self):
         if self.point_output_process is not None:
             self.point_output_process.ExecuteBeforeOutputStep()
+            if self.point_output_process.IsOutputStep():
```

Comment:
As far as I know, the idea behind```OutputProcess::PrintOutput``` is to write the output without checking ```OutputProcess::IsOutputStep```.

```PythonSolver``` already [performs the check] directly in the solution loop once, and then calls ```PrintOutput``` if successful. I don't think it's good practice by @sunethwarna tells me that some people rely on this specific behaviour.

Reply:
Yes, you are right, that was a mistake.

Reply:
I just noticed, there was reasoning behind this .. 
you are right that the `PrintOutput` of `GiDOutputProcess` will only be called if `IsOutputStep` returns true. But analysis_stage won't know about `PointOutputProcess`es inside `GiDOutputProcess` and their output interval will either be checked inside the `PrintOutput` of `GiDOutputProcess` or not at all and the interval could be different.